sweepbatcher/sweep_batcher_test.go

```diff
@@ -1029,3 +1030,38 @@ func TestGetFeePortionForSweep(t *testing.T) {
 		})
 	}
 }
+
+// TestRestoringEmptyBatch tests that the batcher can be restored with an empty
+// batch.
+func TestRestoringEmptyBatch(t *testing.T) {
+	defer test.Guard(t)()
+
+	lnd := test.NewMockLnd()
+	ctx, cancel := context.WithCancel(context.Background())
+
+	store := loopdb.NewStoreMock(t)
+
+	batcherStore := NewStoreMock()
+	_, err := batcherStore.InsertSweepBatch(ctx, &dbBatch{})
+	require.NoError(t, err)
+
+	batcher := NewBatcher(lnd.WalletKit, lnd.ChainNotifier, lnd.Signer,
+		testMuSig2SignSweep, nil, lnd.ChainParams, batcherStore, store)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+		err = batcher.Run(ctx)
+	}()
+
+	// Wait for the batcher to be initialized.
+	<-batcher.initDone
+
+	// Now make it quit by canceling the context.
+	cancel()
+	wg.Wait()
```

Comment:
could go a bit further

add a sweep and verify that we now have 1 batch in memory, and 2 in storage

Reply:
done

Reply:
I've changed the test a bit since the dormant batch is now dropped from the db.